src/metax_api/tests/api/rest/base/views/directories/read.py

```diff
@@ -264,6 +264,26 @@ def test_retrieve_requested_file_and_directory_fields_only(self):
         self.assertEqual(len(response.data['directories'][0].keys()), 1)
         self.assertEqual('id' in response.data['directories'][0], True)
 
+    def test_not_retrieving_not_allowed_directory_fields(self):
+        from metax_api.api.rest.base.serializers import DirectorySerializer, FileSerializer
+
+        allowed_dir_fields = set(DirectorySerializer.Meta.fields)
+        allowed_file_fields = set(FileSerializer.Meta.fields)
+
+        response = self.client.get('/rest/directories/3/files?file_fields=parent,id&directory_fields=;;drop db;')
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertTrue(any(field in response.data['files'][0].keys() for field in allowed_file_fields))
+        self.assertTrue(any(field in response.data['directories'][0].keys() for field in allowed_dir_fields))
+
+        response = self.client.get('/rest/directories/3/files?file_fields=parent&directory_fields=or')
+        self.assertEqual(response.status_code, status.HTTP_500_INTERNAL_SERVER_ERROR)
```

Comment:
A crash probably shouldn't be an expected result, but a 400? As in something was wrong in the parameters the user provided? Its plausible the API user just has a typo in a field name for example or mistakens a field for something else, and will be confused why they are getting a crash without input what they are doing wrong. In case there were 0 valid fields to retrieve, maybe return 400 with info "got invalid fields, choose from the following fields: x, y..." or something along those lines?